### PR TITLE
Extract range checks from riscv into std machines

### PR DESCRIPTION
**Pull request by Schaeff**
Also, allow vms to have links outside instructions.

### CODE REVIEW

**Comment by leonardoalt** [comment]
has conflicts

**Comment by Schaeff** [comment]
will fix

**Review by leonardoalt — APPROVED** [review]
Nice!

File: riscv/src/code_gen.rs [hunk 1]
```diff
@@ -735,9 +732,7 @@ fn memory(with_bootloader: bool) -> String {
 
     // ============== memory instructions ==============
 
-    let up_to_three: col = |i| i % 4;
-    let six_bits: col = |i| i % 2**6;
-    /// Loads one word from an address V = val(X) + Y, where V can be between 0 and 2**33 (sic!),
+    /// Loads one word from an address Y, where Y can be between 0 and 2**33 (sic!),
```
Comment (Schaeff) [inline riscv/src/code_gen.rs:735]: This is wrong

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**